internal/command/init.go

```diff
@@ -1031,23 +1031,87 @@ func (c *InitCommand) backendConfigOverrideBody(flags arguments.FlagNameValueSli
 			flushVals() // deal with any accumulated individual values first
 			mergeBody(newBody)
 		} else {
+			// The flag value is setting a single attribute's value
 			name := item.Value[:eq]
 			rawValue := item.Value[eq+1:]
-			attrS := schema.Attributes[name]
-			if attrS == nil {
+
+			splitName := strings.Split(name, ".")
+			isNested := len(splitName) > 1
+
+			var value cty.Value
+			var valueDiags tfdiags.Diagnostics
+			switch {
+			case !isNested:
+				// The flag item is overriding a top-level attribute
+				attrS := schema.Attributes[name]
+				if attrS == nil {
+					diags = diags.Append(tfdiags.Sourceless(
+						tfdiags.Error,
+						"Invalid backend configuration argument",
+						fmt.Sprintf("The backend configuration argument %q given on the command line is not expected for the selected backend type.", name),
+					))
+					continue
+				}
+
+				value, valueDiags = configValueFromCLI(item.String(), rawValue, attrS.Type)
+				diags = diags.Append(valueDiags)
+				if valueDiags.HasErrors() {
+					continue
+				}
+
+				// Synthetic values are collected as we parse each flag item
+				synthVals[name] = value
+			case isNested:
+				// The flag item is overriding a nested attribute
+				// e.g. assume_role.role_arn in the s3 backend
+				// We assume a max nesting-depth of 1 as s3 is the only
+				// backend that contains nested fields.
+
+				parentName := splitName[0]
+				nestedName := splitName[1]
+				parentAttr := schema.Attributes[parentName]
```

Comment:
There is a path to panicking here, if the schema does not have `parentName`

e.g `"-backend-config=test_nesting_single_not_exists.child=foobar"`